command/agent/auth/auth.go

```diff
@@ -326,15 +326,41 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 	}
 }
 
-// calculateBackoff determines a new backoff duration that is roughly twice
-// the previous value, capped to a max value, with a measure of randomness.
-func calculateBackoff(previous, max time.Duration) time.Duration {
-	maxBackoff := 2 * previous
-	if maxBackoff > max {
-		maxBackoff = max
+// agentBackoff tracks exponential backoff state.
+type agentBackoff struct {
+	max     time.Duration
+	current time.Duration
+}
+
+func newAgentBackoff(max time.Duration) *agentBackoff {
+	if max <= 0 {
+		max = defaultMaxBackoff
+	}
+
+	return &agentBackoff{
+		max:     max,
+		current: initialBackoff,
+	}
+}
+
+// next determines the next backoff duration that is roughly twice
+// the current value, capped to a max value, with a measure of randomness.
+func (b *agentBackoff) next() {
+	maxBackoff := 2 * b.current
+
+	if maxBackoff > b.max {
+		maxBackoff = b.max
 	}
 
 	// Trim a random amount (0-25%) off the doubled duration
 	trim := rand.Int63n(int64(maxBackoff) / 4)
-	return maxBackoff - time.Duration(trim)
+	b.current = maxBackoff - time.Duration(trim)
+}
+
+func (b *agentBackoff) reset() {
+	b.current = initialBackoff
+}
+
+func (b agentBackoff) String() string {
+	return fmt.Sprintf("%.2fs", b.current.Seconds())
```

Comment:
You could `return b.current.String()` and let the stringer implementation from time.Duration handle the formatting.

Reply:
Updated in https://github.com/hashicorp/vault/pull/11033/commits/eba830eb62fbb30881235183bac4aef5357e7b36. I did this since I prefer fewer sig figs... `backoff=1.91s` instead of `backoff=8.682942552s`. Looks like `time.Truncate` does that job just as well.